tests/subsys/suit/manifest_variables/src/main.c

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#include <zephyr/ztest.h>
+#include <suit_manifest_variables.h>
+#include <suit_storage.h>
+
+static void test_suite_before(void *f)
+{
+	suit_plat_err_t ret = suit_storage_init();
+
+	zassert_equal(ret, SUIT_PLAT_SUCCESS, "Failed to initialize SUIT storage module, error: %i",
+		      ret);
+}
+
+ZTEST_SUITE(mfst_variables_tests, NULL, NULL, test_suite_before, NULL, NULL);
+
+ZTEST(mfst_variables_tests, test_var_nvm_OK)
+{
+	suit_plat_err_t ret = SUIT_PLAT_SUCCESS;
+
+	for (size_t i = 0; i < CONFIG_SUIT_MANIFEST_VARIABLES_NVM_COUNT; i++) {
+		uint32_t id = CONFIG_SUIT_MANIFEST_VARIABLES_NVM_BASE_ID + i;
+		/* size of NVM based variable is limited to 8 bits
+		 */
+		uint32_t value = id & 0xFF;
+
+		ret = suit_mfst_var_set(id, value);
+		zassert_equal(ret, SUIT_PLAT_SUCCESS,
+			      "suit_mfst_var_set failed, id: %d, value: %d, error %i", id, value,
+			      ret);
+	}
+
+	for (size_t i = 0; i < CONFIG_SUIT_MANIFEST_VARIABLES_NVM_COUNT; i++) {
+		uint32_t id = CONFIG_SUIT_MANIFEST_VARIABLES_NVM_BASE_ID + i;
+		uint32_t expected_value = id & 0xFF;
+		uint32_t value = 0;
+		uint32_t access_mask = 0;
+
+		ret = suit_mfst_var_get_access_mask(id, &access_mask);
+		zassert_equal(ret, SUIT_PLAT_SUCCESS,
```

Comment:
Generally speaking - it might be worth to add an assertion on the returned `access_mask` value.